internal/reader/ccip.go

```diff
@@ -523,10 +567,52 @@ func (r *CCIPChainReader) Sync(ctx context.Context) (bool, error) {
 				Name:    consts.ContractNameOnRamp,
 			},
 		}); err != nil {
-			return false, fmt.Errorf("bind onRamp: %w", err)
+			return fmt.Errorf("bind onRamp: %w", err)
 		}
 	}
 
+	return nil
+}
+
+func (r *CCIPChainReader) bindNonceManager(ctx context.Context) error {
+	staticConfig, err := r.getOfframpStaticConfig(ctx)
+	if err != nil {
+		return fmt.Errorf("get offramp static config: %w", err)
+	}
+
+	if _, ok := r.contractReaders[r.destChain]; !ok {
+		r.lggr.Debugw("skipping nonce manager, dest chain not configured for this deployment",
+			"destChain", r.destChain)
+		return nil
+	}
+
+	// Bind the nonceManager contract address to the reader.
+	// If the same address exists -> no-op
+	// If the address is changed -> updates the address, overwrites the existing one
+	// If the contract not binded -> binds to the new address
+	if err := r.contractReaders[r.destChain].Bind(ctx, []types.BoundContract{
+		{
+			Address: typeconv.AddressBytesToString(staticConfig.NonceManager, uint64(r.destChain)),
+			Name:    consts.ContractNameNonceManager,
+		},
+	}); err != nil {
+		return fmt.Errorf("bind nonce manager: %w", err)
+	}
+
+	return nil
+}
+
+func (r *CCIPChainReader) Sync(ctx context.Context) (bool, error) {
```

Comment:
@dimkouv why the `(bool, error)` return signature? I feel like `error` is enough?